src/components/TopBar/CompleteButton.js

```diff
@@ -3,30 +3,22 @@
 import React from "react";
 import PropTypes from "prop-types";
 import tickIcon from "../../assets/icons/tick_white.svg";
-import { Link } from "react-router-dom";
+import { Link } from "react-router-dom"; /* eslint-disable-line */
 
+// <CompleteButton> :: can be called in two ways:
+// 1. As an onClick() (used by <RevisionPage>)
+// 2. As a <Link to> using /this.props.exam/this.props.station (used by <AddTile> and <NewCase>)
 export default class CompleteButton extends React.Component {
   render() {
+    const { onClick, exam, station } = this.props;
+    const link = !onClick ? `/${exam}/${station}` : ``;
     return (
       <React.Fragment>
-        <Link
-          to={
-            this.props.exam ? `/${this.props.exam}/${this.props.station}` : ""
-          }
-        >
-          <button
-            id="complete"
-            data-testid="complete"
-            onClick={() => this.props.submitCase()}
-            className={
-              this.props.tickDisplayed
-                ? `tick--displayed`
-                : `tick--not-displayed`
-            }
-          >
-            <img src={tickIcon} />
-          </button>
-        </Link>
+        {link && `<Link to=${link}>`}
+        <button id="complete" data-testid="complete" onClick={() => onClick()}>
+          <img src={tickIcon} />
+        </button>
+        {link && `</Link>`}
```

Comment:
I tested it out, and the complete button still doesn't function properly on the add-station or add-case flow. We can figure it out after this PR, if you like :)

Reply:
Infurating :scream: